### ISSUE TITLE
can't init SPI?

### ISSUE THREAD

**Issue by Sbatushe** [picnoir/ultimate-writer#8]
Hello, I've followed the tutorial on how to install ultimate-typewriter and kept track of all of my commands. this is my setup:
```
#Logkeys
$ sudo apt install build-essential autotools-dev autoconf kbd
$ git clone https://github.com/kernc/logkeys
$ cd logkeys
$ ./autogen.sh
$ cd build
$ ../configure
$ make
$ sudo make install 
$ cd

#bcm2835
$ curl http://www.airspayce.com/mikem/bcm2835/bcm2835-1.71.tar.gz --output lib.tar.gz
$ tar -xvf lib.tar.gz bcm2835-1.71/
$ rm lib.tar.gz
$ cd bcm2835-1.71/
$ ./configure
$ make
$ sudo make check
$ sudo make install
$ cd

#Ultimate-typewriter
$ git clone https://github.com/NinjaTrappeur/ultimate-writer 
$ cd ultimate-writer 
$ make
$ sudo make install
$ sudo systemctl enable ultimatewriter
$ sudo systemctl start ultimatewriter
```

Installing process was straightforward, got no errors. However, when i run _utlimatewriter_ i get a message saying that seems like SPI interface is not setup correctly. Maybe i did something wrong? I used `sudo raspi-config` to enable the SPI interface. I connected my 7.5 inch e-ink display as showed [here](https://m.media-amazon.com/images/I/71rfjUETTjL._AC_SL1452_.jpg).

**Comment by picnoir** [comment]
Hey,

I'm sorry, I won't be able to help you with this :(

It's been 4 years since I stopped working on this project and I pillaged a lot of its component for other stuff. I don't have a working prototype anymore :(

Do you have the same issue using this tag https://github.com/NinjaTrappeur/ultimate-writer/releases/tag/0.1 ?

**Comment by Sbatushe** [comment]
Yes, i tried both with release and current branch. I will try to solve this problem and I'll document everything here :)

**Comment by za3k** [comment]
Did you reboot after raspi-config? Do you have the two switches set as indicated on the HAT? Can you include the message?

I'm working through this now and I am not getting such a message. (I am encountering problems after this step, will have a PR at some point)

**Comment by Sbatushe** [comment]
Which switches? my HAT is like [that](https://www.i-diskhouse.shop/wp-content/uploads/2021/06/Waveshare-7-5inch-E-Ink-display-HAT-for-Raspberry-Pi-800-480-Black-White-two-colors.jpg). I started to develop my own project because i was unable to solve this problem. I don't know if the problem could be my display, because it's a 7.5 inch.

**Comment by za3k** [comment]
There are two physical switched on the HAT, like in the picture. I also have the 7.5 inch display

**Comment by za3k** [comment]
I can probably help on this, if you include the *output* of running the above. Set your switches on the HAT to "B0".

**Comment by dwehrmann** [comment]
I had the same problem, too:
_Cannot init SPI interface. Is it activated in raspi-config?_
solved it with `sudo ultimatewriter` – it might work for you, too.
